core/evm.go

```diff
@@ -249,6 +252,56 @@ func MigrateDelegationsFn(ref *block.Header, chain ChainContext) vm.MigrateDeleg
 	}
 }
 
+// calculate the gas for migration; no checks done here similar to other functions
+func CalculateMigrationGasFn(chain ChainContext) vm.CalculateMigrationGasFunc {
+	return func(db vm.StateDB, migrationMsg *stakingTypes.MigrationMsg, homestead bool, istanbul bool) (uint64, error) {
+		var gas uint64 = 0
+		delegations, err := chain.ReadDelegationsByDelegator(migrationMsg.From)
+		if err != nil {
+			return 0, err
+		}
+		for i := range delegations {
+			delegationIndex := &delegations[i]
+			wrapper, err := db.ValidatorWrapper(delegationIndex.ValidatorAddress, true, false)
+			if err != nil {
+				return 0, err
+			}
+			foundDelegation := &wrapper.Delegations[delegationIndex.Index]
```

Comment:
The index out of bound check is removed?

Reply:
Good catch; added in the gas calculation as well now.